src/System.Private.CoreLib/src/System/Runtime/InteropServices/Marshal.cs

```diff
@@ -861,7 +861,7 @@ public static IntPtr GetHINSTANCE(Module m)
 
             if (rtModule == null)
```

Comment:
If we want to continue on this path, this can be simplified to (the unmanaged implementation would always returned `-1` for :

```
public static IntPtr GetHINSTANCE(Module m)
{
    if (m == null)
    {
        throw new ArgumentNullException(nameof(m));
    }

    if (m is RuntimeModule rtModule)
    {
        return GetHINSTANCE(rtModule.GetNativeHandle());
    }

    return (IntPtr)(-1);
}
```